turbopack/crates/turbo-tasks-auto-hash-map/src/map.rs

```diff
@@ -900,6 +901,25 @@ where
     }
 }
 
+impl<K, V, H, const I: usize> ShrinkToFit for AutoMap<K, V, H, I>
+where
+    K: Eq + Hash,
+    V: Eq,
+    H: BuildHasher + Default,
+{
+    fn shrink_to_fit(&mut self) {
+        match self {
+            AutoMap::List(list) => list.shrink_to_fit(),
+            AutoMap::Map(map) => {
+                hashbrown::HashMap::shrink_to_fit(map);
+
+                if map.len() < MIN_HASH_SIZE {
+                    self.convert_to_list();
+                }
```

Comment:
No need to shrink though if it's getting converted anyway?
```suggestion
                if map.len() < MIN_HASH_SIZE {
                    self.convert_to_list();
                } else {
                    hashbrown::HashMap::shrink_to_fit(map);
                }
```